ui/widgets/src/main/java/com/appcoins/wallet/ui/widgets/TransactionsComposable.kt

```diff
@@ -0,0 +1,150 @@
+import androidx.compose.ui.unit.dp
+import coil.compose.SubcomposeAsyncImage
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_blue_secondary
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_dark_grey
+import com.appcoins.wallet.ui.common.theme.WalletColors.styleguide_light_grey
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TransactionCard(
+  icon: Int?,
+  appIcon: String?,
+  title: String,
+  description: String?,
+  amount: String?,
+  convertedAmount: String?,
+  subIcon: Int?,
+  onClick: () -> Unit,
+  textDecoration: TextDecoration
+) {
+  Card(
+    modifier = Modifier.fillMaxWidth(),
+    colors = CardDefaults.cardColors(containerColor = styleguide_blue_secondary),
+    onClick = onClick
+  ) {
+    Row(
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically,
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(vertical = 8.dp, horizontal = 16.dp)
+    ) {
+      Row(verticalAlignment = Alignment.CenterVertically) {
+        TransactionItem(icon, appIcon, subIcon)
+        Spacer(modifier = Modifier.padding(14.dp))
+        Column(modifier = Modifier.widthIn(0.dp, 168.dp)) {
+          Text(
+            text = title,
+            fontWeight = FontWeight.Bold,
+            color = styleguide_light_grey,
+            style = MaterialTheme.typography.bodyMedium,
+            maxLines = 1,
+            overflow = TextOverflow.Ellipsis
+          )
+          if (description != null) Text(
+            text = description,
+            color = styleguide_dark_grey,
+            style = MaterialTheme.typography.bodySmall,
+            maxLines = 2,
+            overflow = TextOverflow.Ellipsis
+          )
+        }
+      }
+      Column(horizontalAlignment = Alignment.End) {
+        if (amount != null) Text(
+          text = amount,
+          fontWeight = FontWeight.Bold,
+          style = MaterialTheme.typography.bodyMedium,
+          textAlign = TextAlign.End,
+          color = styleguide_light_grey,
+          maxLines = 1,
+          overflow = TextOverflow.Ellipsis,
+          textDecoration = textDecoration
+        )
+        if (convertedAmount != null) Text(
+          text = convertedAmount,
+          color = styleguide_dark_grey,
+          style = MaterialTheme.typography.bodySmall,
+          textAlign = TextAlign.End,
+          maxLines = 1,
+          overflow = TextOverflow.Ellipsis,
+          textDecoration = textDecoration
+        )
+      }
+    }
+  }
+}
+
+@Composable
+fun TransactionItem(icon: Int?, appIcon: String?, subIcon: Int?) {
+  Box(contentAlignment = Alignment.BottomEnd) {
+    if (icon != null) Icon(
+      painter = painterResource(id = icon),
+      contentDescription = null,
+      tint = Color.Unspecified,
+      modifier = Modifier.size(40.dp)
+    ) else {
+      SubcomposeAsyncImage(
+        model = appIcon,
+        contentDescription = "Game Icon",
```

Comment:
hardcoded description might be an issue in alternative languages

Reply:
you're right, my mistake, I will change it